pkg/bundle/internal/target/target.go

```diff
@@ -242,6 +254,15 @@ func (r *Reconciler) needsUpdate(ctx context.Context, kind Kind, log logr.Logger
 		needsUpdate = true
 	}
 
+	// Get old password hash annotations.
+	// Calculate new password hash annotation with changed password.
+	// Compare old password hash annotations with new ones.
+	if bundle.Spec.Target.AdditionalFormats != nil {
+		if !isHashAnnotationEqual(getOldHashAnnotations(obj.GetAnnotations(), bundle.Spec.Target.AdditionalFormats), calcNewHashAnnotations(bundle.Spec.Target.AdditionalFormats)) {
```

Comment:
Is it required to calculate the desired annotations again? Why can't you just supply the desired annotations as a single map? Possibly bundled with the existing data hash annotation?

Reply:
Do you want me to make use two static annotation keys like `trust.cert-manager.io/jks-hash` and `trust.cert-manager.io/pksc12-hash` instead using annotation keys based on targets keys?

In that case code would be

```
if obj.GetLabels()[trust.cert-manager.io/jks-hash] != bundle.Name {
		needsUpdate = true
}
if obj.GetLabels()[trust.cert-manager.io/pksc12-hash] != bundle.Name {
		needsUpdate = true
}
```